### PR TITLE
add more data source for auto download

### PR DESCRIPTION
**Pull request by gulou01517514**
# Motivation
data support services, add more datasource class

# Modification
The main thing is datasource auto download functionality

### CODE REVIEW

File: easycv/datasets/detection/data_sources/coco_livs.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        transforms = []
+        for transform in pipeline:
+            if isinstance(transform, dict):
+                transform = build_from_cfg(transform, PIPELINES)
+                transforms.append(transform)
+            elif callable(transform):
+                transforms.append(transform)
+            else:
+                raise TypeError('transform must be callable or a dict')
+        self.pipeline = Compose(transforms)
```
Comment (wenmengzhou) [inline easycv/datasets/detection/data_sources/coco_livs.py:99]: call DetSourceCoco.__init__

File: easycv/datasets/classification/data_sources/caltech.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+        # data classes
+        self.CLASSES = self.caltech256.categories
+
+    def __len__(self):
+        return len(self.caltech256.index)
+
+    def __getitem__(self, idx):
+        # img: HWC, RGB
+        img, label = self.caltech256[idx]
```
[the remark this answers is not in the document]
Comment (gulou01517514) [reply easycv/datasets/classification/data_sources/caltech.py:92]: yes, return PILImage torch.data.Caltech256 for classifcation

File: easycv/datasets/detection/data_sources/coco_livs.py (hunk 1 above)
Comment (gulou01517514) [reply easycv/datasets/detection/data_sources/coco_livs.py:99]: no , coco is different from livs in init , some places are handled differently

File: docs/source/data_hub.md [hunk 3]
```diff
@@ -24,52 +24,50 @@ Before using dataset, please read the [LICENSE](docs/source/LICENSE) file to lea
 
 ## Classification data
 
-| Name                                                         | Field  | Describtion                                                  | Download                                                     | Dataset API support                                 | Licence                                 |
-| ------------------------------------------------------------ | ------ | ------------------------------------------------------------ | ------------------------------------------------------------ | --------------------------------------- | --------------------------------------- |
+| Name                                                         | Field  | Describtion                                                  | Download                                                     | Dataset API support                     | Licence                                 |
```
Comment (wenmengzhou) [inline docs/source/data_hub.md:27]: dataset API支持的需要给出支持的类型，在后面新增一个章节给出使用示例

**Review by wenmengzhou — APPROVED** [review]

**Pull request merged**